### PR TITLE
init.d script -- fix issue #3803

### PR DESCRIPTION
**Pull request by KoeSystems**
This bug was introduced in PR 3115.

### CODE REVIEW

File: scripts/init.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ PIDFILE=/var/run/influxdb/influxd.pid
 PIDDIR=`dirname $PIDFILE`
 if [ ! -d "$PIDDIR" ]; then
     mkdir -p $PIDDIR
-    chown $GROUP:$USER $PIDDIR
+    chown $USER:$GROUP $PIDDIR
```
Comment (KoeSystems) [inline scripts/init.sh]: Thanks to @dgnorton in PR 3115 I realise there's a bug here.

**Comment by toddboom** [comment]
Well, init scripts aren't my forte, but this all seems to make sense to me. What do you think, @otoolep?

File: scripts/init.sh [hunk 2]
```diff
@@ -62,6 +62,7 @@ fi
 
 if [ ! -f "$STDOUT" ]; then
     mkdir -p $(dirname $STDOUT)
+    chown $USER:$GROUP $(dirname $STDOUT)
```
Comment (otoolep) [inline scripts/init.sh]: Surely not. If `STDOUT` (and `STDERR`) are not changed from the defaults, this is changing the owner and group of `/dev`.

Comment (otoolep) [inline scripts/init.sh]: I.e. it makes no sense to do this, right?

`chown influxdb:influxdb /dev`

File: scripts/init.sh (hunk 1 above)
Comment (otoolep) [inline scripts/init.sh]: +1 on this specific change.

File: scripts/init.sh [hunk 3]
```diff
@@ -151,7 +153,7 @@ case $1 in
         if which start-stop-daemon > /dev/null 2>&1; then
             start-stop-daemon --chuid $GROUP:$USER --start --quiet --pidfile $PIDFILE --exec $DAEMON -- -pidfile $PIDFILE -config $CONFIG $INFLUXD_OPTS >>$STDOUT 2>>$STDERR &
         else
-            nohup $DAEMON -pidfile $PIDFILE -config $CONFIG $INFLUXD_OPTS >>$STDOUT 2>>$STDERR &
+            su -c "nohup $DAEMON -pidfile $PIDFILE -config $CONFIG $INFLUXD_OPTS >>$STDOUT 2>>$STDERR &" $USER
```
Comment (otoolep) [inline scripts/init.sh]: +1 on this specific change.

**Comment by otoolep** [comment]
2 of the changes make sense. The other 2 scare me!

**Comment by otoolep** [comment]
Please let me know if I am missing something here.

File: scripts/init.sh (hunk 2 above)
Comment (KoeSystems) [inline scripts/init.sh]: Opps, that's true, I was trying to fix the corner case when you change from the default user influxdb to another user and the file/directory already exist.

I will revert this change.

**Comment by KoeSystems** [comment]
@otoolep @toddboom It seems there is something wrong with the CI test.
```
github.com/influxdb/influxdb/uuid	[no test files] bash circle-test.sh returned exit code 1
```
You can see the same error in other PR

**Comment by otoolep** [comment]
Thanks @KoeSystems -- can you squash the 3 changes into 1?

**Comment by otoolep** [comment]
Closed in error.

**Comment by otoolep** [comment]
Thanks @KoeSystems -- merging now.

**Pull request merged**